client/client-multi/client-c/src/kaa/platform-impl/common/kaa_client_properties.c

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2014-2016 CyberVision, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <platform/kaa_client_properties.h>
+#include <platform/stdio.h>
+#include <string.h>
+#include <platform/mem.h>
+#include "utilities/kaa_mem.h"
+
+#define KAA_DEFAULT_WORKING_DIRECTORY "./"
+
+static kaa_client_props_t kaa_client_props;
+
+kaa_error_t kaa_client_props_set(const kaa_client_props_t *self)
+{
+    if (self == NULL) {
+        kaa_client_props.working_directory = KAA_CALLOC(1, sizeof(KAA_DEFAULT_WORKING_DIRECTORY) + 1);
```

Comment:
The `sizeof` operator applied to string literals returns size including the NULL-termination byte, no need to add 1 here.